nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearch5.java

```diff
@@ -227,6 +227,7 @@ public void process(final InputStream in) throws IOException {
                         session.transfer(flowFile, REL_FAILURE);
 
                     } else {
+                        session.getProvenanceReporter().send(flowFile, response.remoteAddress().getAddress());
```

Comment:
This will only emit provenance event if "response.hasFailures" is true. If there are not failures this won't get hit.